test_communication/test/subscriber_py.py

```diff
@@ -42,17 +42,20 @@ def listener_cb(msg, received_messages, expected_msgs):
         raise RuntimeError('received unexpected message %r' % msg)
 
 
-def listener(message_name, rmw_implementation, number_of_cycles):
+def listener(message_name, number_of_cycles):
+    from message_fixtures import get_test_msg
     import rclpy
-    from rclpy.qos import qos_profile_default
     from rclpy.impl.rmw_implementation_tools import select_rmw_implementation
-    from message_fixtures import get_test_msg
+    from rclpy.impl.rmw_implementation_tools import get_rmw_implementations
+    from rclpy.qos import qos_profile_default
 
     message_pkg = 'test_communication'
     module = importlib.import_module(message_pkg + '.msg')
     msg_mod = getattr(module, message_name)
 
-    select_rmw_implementation(rmw_implementation)
+    rmw_implementations = get_rmw_implementations()
+    assert(os.environ['RCLPY_IMPLEMENTATION'] in rmw_implementations)
```

Comment:
would it make sense to remove this check (and the corresponding import) as it's handled by `select_rmw_implementation`  with a specific exception (https://github.com/ros2/rclpy/blob/886ae658a3dcc3aafc8e7ceb697161df2faac846/rclpy/rclpy/impl/rmw_implementation_tools.py#L66) ?